pkg/injection/namespace/initgeneration/initgeneration.go

```diff
@@ -63,16 +63,16 @@ func (g *InitGenerator) GenerateForNamespace(ctx context.Context, dk dynatracev1
 		Data: data,
 		Type: corev1.SecretTypeOpaque,
 	}
-	secretQuery := k8ssecret.NewQuery(ctx, g.client, g.apiReader, log)
+	secretQuery := k8ssecret.Query(g.client, g.apiReader, log)
 
-	err = secretQuery.CreateOrUpdate(*secret)
+	_, err = secretQuery.CreateOrUpdate(ctx, secret)
 
 	return errors.WithStack(err)
 }
 
 // GenerateForDynakube creates/updates the init secret for EVERY namespace for the given dynakube.
 // Used by the dynakube controller during reconcile.
-func (g *InitGenerator) GenerateForDynakube(ctx context.Context, dk *dynatracev1beta2.DynaKube) error {
+func (g *InitGenerator) GenerateForDynakube(ctx context.Context, dk *dynakube.DynaKube) error {
```

Comment:
Would it make sense to align the interface here as well?
- `dynakube.Dynakube` parameter for `GenerateForNamespace`
- `*dynakube.Dynakube` parameter for `GenerateForDynakube`